explanation/intro-to/ebpf.md

```diff
@@ -0,0 +1,122 @@
+(you can also inspect those as you would inspect the `.bt` files).
+
+You will also find several examples for `bpftrace` in
+`/usr/share/doc/bpftrace/examples/` and for `bpfcc-tools` in
+`/usr/share/doc/bpfcc-tools/examples/`.
+
+For instance,
+
+```
+# bashreadline.bt
+```
+
+will print bash commands for all running bash shells in your system. Note that
+you need to run these scripts as root. You will also notice that there is also
+a `bashreadline-bpfcc` tool available from `bpfcc-tools`. Both of them provide
+similar features. The former is implemented in python with BCC while the latter
+is a bpftrace script, as described above. You will see that many of these tools
+have both a `-bpfcc` and a `.bt` version. Do read their manpages (and perhaps
+the scripts) to choose which suits you best.
+
+On a more practical application, let's say you want to verify which bios is
+used with particular commandlines of qemu. You could definitely use `strace` to
+do so, but that would likely add some overhead to the investigation process.
+Instead, you can simply use `opensnoop.bt` to trace `open()` syscalls:
+
+```
+# While running one
+$ sudo /usr/sbin/opensnoop-bpfcc -TF |& tee file-qemu-seabios.txt
+# While running the other
+sudo /usr/sbin/opensnoop-bpfcc -TF |& tee file-qemu-qboot.txt
+# Result comparison is super simple
+grep -Hn qemu file-qemu-* | grep share
+file-qemu-qboot.txt:972:1.699723000   3413787 qemu-system-x86    12   0 /usr/share/qemu/qboot.rom
+file-qemu-qboot.txt:973:1.699904000   3413787 qemu-system-x86    12   0 /usr/share/qemu/qboot.rom
+file-qemu-qboot.txt:1032:1.718886000   3413787 qemu-system-x86    12   0 /usr/share/qemu/kvmvapic.bin
+file-qemu-qboot.txt:1033:1.718944000   3413787 qemu-system-x86    12   0 /usr/share/qemu/linuxboot_dma.bin
+file-qemu-qboot.txt:1041:1.728519000   3413787 qemu-system-x86    12   0 /usr/share/seabios/vgabios-stdvga.bin
+file-qemu-qboot.txt:1042:1.728626000   3413787 qemu-system-x86    12   0 /usr/share/seabios/vgabios-stdvga.bin
+file-qemu-qboot.txt:1046:1.764491000   3413787 qemu-system-x86    12   0 /usr/share/terminfo/x/xterm-256color
+file-qemu-seabios.txt:888:1.617490000   3413913 qemu-system-x86    12   0 /usr/share/seabios/bios-256k.bin
+file-qemu-seabios.txt:889:1.617706000   3413913 qemu-system-x86    12   0 /usr/share/seabios/bios-256k.bin
+file-qemu-seabios.txt:904:1.662015000   3413913 qemu-system-x86    12   0 /usr/share/qemu/kvmvapic.bin
+file-qemu-seabios.txt:905:1.662068000   3413913 qemu-system-x86    12   0 /usr/share/qemu/linuxboot_dma.bin
+file-qemu-seabios.txt:911:1.681084000   3413913 qemu-system-x86    12   0 /usr/share/seabios/vgabios-stdvga.bin
+file-qemu-seabios.txt:912:1.681191000   3413913 qemu-system-x86    12   0 /usr/share/seabios/vgabios-stdvga.bin
+file-qemu-seabios.txt:928:1.712340000   3413913 qemu-system-x86    12   0 /usr/share/terminfo/x/xterm-256color
+```
+
+Which shows one uses `qboot.rom` and the other uses `bios-256k.bin`.
+
```

Comment:
I'd put a subheading called "conclusion" or something like that

Reply:
Done.